### PR TITLE
[MapPlayground] Switch tooltips to a new MapTooltip component + other enhancements

### PR DESCRIPTION
**Pull request by jshoe**
### Description
- Standardize on onMouseEnter/onMouseLeave
- Standardize on lat/lng instead of lat/lon
- Update the MapTiler style ID to a new one created by Design
- Design said we won't have a foreseeable use for persistent on-click popups so I took them out of MapPlayground for now.
- Remove radius from the CircleMarker per Design
- Change the tooltips to a new MapTooltip component with an updated style. Make the tooltip feel "naturally sticky" so that it won't close if you hover over the tooltip instead of the marker, and so that it has a slight delay before closing.
- TODO later: Have a circle radius only on selection, adding linking to the tooltip title, etc.

### Demo
Screen Shot 2019-04-25 at 2 01 10 PM
Apr-25-2019 14-03-08

### Test and Reproduce
- Pull the branch, retrieve MAPTILER_API_KEY from Parameter Store, start locally with `MAPTILER_API_KEY=key_here docker-compose up web`
- Visit http://localhost:3000/locations/map_playground
- Hover over bubble markers and look at tooltip style. Move around bubbles and observe that it won't close while over a tooltip but will after you move away.

### CODE REVIEW

File: app/assets/src/components/views/discovery/mapping/BaseMap.jsx [hunk 1]
```diff
@@ -27,10 +27,11 @@ class BaseMap extends React.Component {
   };
 
   render() {
-    const { mapTilerKey, hoverTooltip, markers, popups } = this.props;
+    const { mapTilerKey, tooltip, markers, popups } = this.props;
     const { viewport } = this.state;
 
-    const styleID = "6041e288-ea0e-4a64-b0f6-78172d56a657";
+    // MapTiler map name: "2019-04-24"
+    const styleID = "f0e7922a-43cf-4dc5-b598-17ae1f56d2f4";
```
Comment (MarkAZhang) [inline app/assets/src/components/views/discovery/mapping/BaseMap.jsx]: Nit: can we move this to the top as a constant? Maybe "IDSEQ_MAP_STYLE_ID".

File: app/assets/src/components/views/discovery/mapping/MapPlayground.jsx [hunk 2]
```diff
@@ -83,102 +81,60 @@ class MapPlayground extends React.Component {
...
+  handleMarkerMouseLeave = () => {
+    // Flag the tooltip to close after a timeout, which could be unflagged by another event (entering a marker or tooltip).
+    this.setState({ tooltipShouldClose: true });
 
-  closePopup = popupInfo => {
-    this.setState({
-      popups: reject({ index: popupInfo.index }, this.state.popups)
-    });
+    const delayTimeMs = 2000;
+    setTimeout(() => {
```
Comment (MarkAZhang) [inline app/assets/src/components/views/discovery/mapping/MapPlayground.jsx]: Nice.

File: app/assets/src/components/views/discovery/mapping/MapPlayground.jsx [hunk 3]
```diff
@@ -83,102 +81,60 @@ class MapPlayground extends React.Component {
...
-  };
+  handleMarkerMouseLeave = () => {
+    // Flag the tooltip to close after a timeout, which could be unflagged by another event (entering a marker or tooltip).
+    this.setState({ tooltipShouldClose: true });
 
-  closePopup = popupInfo => {
-    this.setState({
-      popups: reject({ index: popupInfo.index }, this.state.popups)
-    });
+    const delayTimeMs = 2000;
```
Comment (MarkAZhang) [inline app/assets/src/components/views/discovery/mapping/MapPlayground.jsx]: Maybe move this to the top as a constant?

Also, is 2000 too long? Zillow seems to be between 500 and 1000ms.

File: app/assets/src/components/views/discovery/mapping/map_tooltip.scss [hunk 4]
```diff
@@ -0,0 +1,55 @@
+@import "~styles/themes/colors";
+@import "~styles/themes/typography";
+
+.mapTooltip {
+  z-index: 1;
+
+  // Global to override some MapboxGL styling
+  :global(.mapboxgl-popup-content) {
+    $tooltip-border-width: 1px;
```
Comment (MarkAZhang) [inline app/assets/src/components/views/discovery/mapping/map_tooltip.scss]: Nit: Could we move constants to the top of the file, below the imports?

File: app/assets/src/components/views/discovery/mapping/BaseMap.jsx (hunk 1 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/mapping/BaseMap.jsx]: 👍 done

File: app/assets/src/components/views/discovery/mapping/MapPlayground.jsx (hunk 3 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/mapping/MapPlayground.jsx]: 👍 it did seem a little long. shortened to 1000ms

File: app/assets/src/components/views/discovery/mapping/map_tooltip.scss (hunk 4 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/mapping/map_tooltip.scss]: 👍 done

**Pull request merged**